mantle/platform/util.go

```diff
@@ -216,6 +216,64 @@ func GetMachineBootId(m Machine) (string, error) {
 	return strings.TrimSpace(string(stdout)), nil
 }
 
+// GetMachineSoftRebootCount retrieves the systemd SoftRebootsCount (for soft-reboot detection)
+func GetMachineSoftRebootCount(m Machine) (string, error) {
+	stdout, stderr, err := m.SSH("systemctl show --value --property SoftRebootsCount")
+	if err != nil {
+		return "", fmt.Errorf("failed to retrieve SoftRebootsCount: %s: %s: %s", stdout, err, stderr)
+	}
+	return strings.TrimSpace(string(stdout)), nil
+}
+
+// WaitForMachineSoftReboot waits for the machine to soft-reboot using systemd SoftRebootsCount
+func WaitForMachineSoftReboot(m Machine, j *Journal, timeout time.Duration, oldSoftRebootsCount string) error {
+	// For soft-reboot, we don't change the boot ID, but the systemd SoftRebootsCount should increment
+	// We use systemd's SoftRebootsCount property to detect when the system has soft-rebooted
+	c := make(chan error)
+	go func() {
```

Comment:
Completely unrelated to this but just for the future...I am increasingly of the opinion that the way tests should work is for the target machine to run autonomously, and report its status instead of being polled. IOW the kubernetes model, not the ansible model. It's a bit ironic that we ended up with a test system that follows more closely to the latter.

At least the external tests always run in a systemd unit. If we instead had an "agent reports status" model then this poll/retry code wouldn't be necessary. In fact the harness wouldn't even need to be aware of soft reboots in general!

But anyways, the test suite is what it is now.